app/models/user.rb

```diff
@@ -27,7 +27,7 @@ class User < ActiveRecord::Base
   end
 
   def tasks_left
-    tasks.select{ |task| task.status != "accepted" }
+    tasks.joins(student: :term).where(terms: {active: true}).where.not(status: Task.statuses[:accepted])
```

Comment:
Yep, I like it!
Although I'd try to fight nested `where`, it's much better than plain SQL here.